internal/releasesources/release_sources.go

```diff
@@ -0,0 +1,499 @@
+		supportLevelsForMajor, err := h.supportLevelClient.getSupportLevels(majorVersion)
+		if err != nil {
+			return nil, err
+		}
+
+		for majorMinorVersion, supportLevel := range supportLevelsForMajor {
+			supportLevels[majorMinorVersion] = supportLevel
+		}
+	}
+
+	return supportLevels, nil
+}
+
+// setSupportLevels sets support level for given release images as follows:
+// candidate/pre-release releases are considered support level 'beta'.
+// The rest are stable releases, and their support level is set according to the mapping
+// retrieved from the API.
+func (h *releaseSourcesHandler) setSupportLevels(releases []*enrichedReleaseImage, supportLevels ocpVersionSupportLevels) error {
+	for _, release := range releases {
+		if release.SupportLevel != "" {
+			continue
+		}
+
+		isPreRelease, err := common.IsVersionPreRelease(*release.Version)
+		if err != nil {
+			return err
+		}
+
+		if release.channel != nil && *release.channel == OpenshiftReleaseChannelCandidate || *isPreRelease {
+			release.SupportLevel = models.OpenshiftVersionSupportLevelBeta
+			continue
+		}
+
+		majorMinorVersion, err := common.GetMajorMinorVersion(*release.Version)
+		if err != nil {
+			return err
+		}
+
+		if supportLevel, exists := supportLevels[*majorMinorVersion]; exists {
+			release.SupportLevel = supportLevel
+			continue
+		}
+
+		return fmt.Errorf("release %s did not appear in the support level mapping and is not a prerelease or candidate", *release.Version)
+	}
+
+	return nil
+}
+
+// setDefaultReleaseImage labels the latest production release image of the default CPU architecture as default
+func (h *releaseSourcesHandler) setDefaultReleaseImage(releases []*enrichedReleaseImage) error {
+	var latestStableRelease *enrichedReleaseImage
+
+	for _, release := range releases {
+		if *release.CPUArchitecture != common.DefaultCPUArchitecture || release.SupportLevel != models.OpenshiftVersionSupportLevelProduction {
+			continue
+		}
+
+		if latestStableRelease == nil {
+			latestStableRelease = release
+		}
+
+		lessThan, err := common.BaseVersionLessThan(*release.Version, *latestStableRelease.Version)
+		if err != nil {
+			return err
+		}
+
+		if lessThan {
+			latestStableRelease = release
+		}
+	}
+
+	if latestStableRelease == nil {
+		h.log.Debugf("No production release images of CPU architecture %s found", common.DefaultCPUArchitecture)
+		return nil
+	}
+
+	h.log.Debugf("Labeling release %s with CPU architecture %s as default", *latestStableRelease.Version, common.DefaultCPUArchitecture)
+	latestStableRelease.Default = true
+
+	return nil
+}
+
+func (h *releaseSourcesHandler) deleteAllReleases(tx *gorm.DB) error {
+	var count int64
+	err := tx.Model(&models.ReleaseImage{}).Count(&count).Error
+	if err != nil {
+		return err
+	}
+	h.log.Debugf("Truncating all release_images table records. %d records", count)
+	return tx.Exec("TRUNCATE TABLE release_images").Error
+}
+
+func (h *releaseSourcesHandler) syncReleaseImagesWithErr() error {
```

Comment:
I can't return an error with `SyncReleaseImages` because of the way `thread` package is currently functioning, so I  created a wrapper function that can return an error, which is much easier to test with